blueocean-dashboard/src/main/js/components/karaoke/components/Step.jsx

```diff
@@ -130,7 +130,12 @@ export class Step extends Component {
             }}
             />);
         } else if (step.isInputStep) {
-            children = <InputStep {...{ step, key: 'step' }} />;
+            children = (<InputStep 
+                {...{ step, key: 'step' }}
+                pipeline={pipeline}
```

Comment:
why not: 

```
{...{ step, key: 'step', pipeline, branch, run }}
```

Reply:
Sure, I'll make the change, I just really don't like that spread operator, it's just not easy to debug and looks weird, especially when used like this, I'll get used to it eventually

Reply:
and I just found out that the use of the spread operator in object literals is not even part of the spec yet, it is just an experimental proposal still subject to change (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#Spread_in_object_literals). Is it a good idea to use it at all until it becomes part of the spec?

Reply:
I personally prefer the spread operator but you can use the other style as well but IMO you should choose either one and not mix those two.

Reply:
yeah you are right, will change it to one style tomorrow, thanks

Reply:
Spreading out a throwaway object literal in a JSX element is an abomination that we should never be doing, IMO.